gamla/functional.py

```diff
@@ -314,3 +315,32 @@ def update_in(d, keys, func, default=None, factory=dict):
     else:
         inner[k] = func(default)
     return rv
+
+
+@toolz.curry
+def dataclass_replace(
+    attr_name: Text, attr_mutator: Callable[[Any], Any], dataclass_instance
+):
+    return dataclasses.replace(
+        dataclass_instance,
+        **{
+            attr_name: toolz.pipe(
+                dataclass_instance, operator.attrgetter(attr_name), attr_mutator
+            )
+        },
+    )
+
+
+_R = TypeVar("_R")
+_E = TypeVar("_E")
+
+
+@toolz.curry
+def reduce(
+    reducer: Callable[[_R, _E], _R], initial_value: _R, elements: Iterable[_E],
+) -> _R:
+    return toolz.pipe(
```

Comment:
not sure why we need the pipeline here ? isn't this as simple as `return functools.reduce(reducer, elements, initial_value)` ?